thaw_utils/src/dom/ssr_mount_style.rs

```diff
@@ -139,6 +139,19 @@ impl RenderHtml for SSRMountStyle {
         buf.insert_str(marker_loc, &self.context.to_html());
     }
 
+    fn to_html_async_with_buf<const OUT_OF_ORDER: bool>(
+        self,
+        buf: &mut leptos::tachys::ssr::StreamBuilder,
+        position: &mut Position,
+        escape: bool,
+        mark_branches: bool,
+    ) where
+        Self: Sized,
+    {
+        self.children
+            .to_html_async_with_buf::<OUT_OF_ORDER>(buf, position, escape, mark_branches);
+    }
```

Comment:
```suggestion
        buf.with_buf(|buf| {
            let head_loc = buf
                .find("<head>")
                .expect("you are using SSRMountStyleProvider without a <head> tag");
            let marker_loc = buf
                .find(r#"<meta name="thaw-ui-style""#)
                .unwrap_or(head_loc + 6);
            buf.insert_str(marker_loc, &self.context.to_html());
        });
    }
```

Implementing this code resolved the issue I mentioned in #346, but I am unsure it is the proper solution.